cpp/src/io/orc/reader_impl.cu

```diff
@@ -432,10 +425,13 @@ table_with_metadata reader::impl::read(size_type skip_rows,
   // Get a list of column data types
   std::vector<data_type> column_types;
   for (const auto &col : _selected_columns) {
-    auto col_type = to_type_id(
-      _metadata->ff.types[col], _use_np_dtypes, _timestamp_type.id(), _decimals_as_float64);
+    auto col_type = to_type_id(_metadata->ff.types[col], _use_np_dtypes, _timestamp_type.id());
+    auto scale    = (col_type == type_id::DECIMAL64) ? _metadata->ff.types[col].scale : 0;
     CUDF_EXPECTS(col_type != type_id::EMPTY, "Unknown type");
-    column_types.emplace_back(col_type);
+    // Remove this once we support Decimal128 data type
+    CUDF_EXPECTS((col_type != type_id::DECIMAL64) or (_metadata->ff.types[col].precision <= 18),
+                 "Decimal data has precision > 18, Decimal64 data type doesn't support it.");
+    column_types.emplace_back(col_type, -1 * static_cast<int32_t>(scale));
```

Comment:
Scale is not use before this line. Could we do the precision check before computing the scale and also initialize scale to `-_metadata->ff.types[col].scale` ?
Also, a comment on why we use negative ORC scale would be nice.